### ISSUE TITLE
MacOS: wxMessageDialog only shows error icon with Yes / No buttons

### ISSUE THREAD

**Issue by wouterbt** [wxWidgets/wxWidgets#24602]
On MacOS, `wxMessageDialog` uses a native dialog that shows the application icon. When used with the `wxICON_ERROR` style, an error icon (exclamation mark triangle) is displayed with the application icon, but *only* with the `wxYES_NO` style, not with the `wxOK` style. 
![ok](https://github.com/wxWidgets/wxWidgets/assets/20640938/b91fb0b2-d390-4a75-b5fc-6dde92c9a76a)
![yes-no](https://github.com/wxWidgets/wxWidgets/assets/20640938/50949663-39b5-41e1-9e66-f7a9f807d607)
The *expected behaviour* is to always show the error icon when the `wxICON_ERROR` style is present. Although [Apple GUI design guidelines](https://developer.apple.com/design/human-interface-guidelines/alerts#macOS) suggest to use the caution symbol sparingly, they do not prohibit it in alerts with only an OK button.

Example code producing the issue:
```c++
#include <wx/wx.h>

class MyApp : public wxApp {
public:
    bool OnInit() {
        wxMessageDialog dlg1(NULL, "text", "Error", wxICON_ERROR | wxOK);
        dlg1.ShowModal();
        dlg1.Destroy();

        wxMessageDialog dlg2(NULL, "text", "Error", wxICON_ERROR | wxYES_NO);
        dlg2.ShowModal();
        dlg2.Destroy();

        return false;
    }
};

wxIMPLEMENT_APP(MyApp);
```

**Comment by vadz** [comment]
This is due to this code here:

https://github.com/wxWidgets/wxWidgets/blob/7f5ca30e3eebbdb85222fbe329a5100b4da65804/src/osx/cocoa/msgdlg.mm#L39-L42

so it's done intentionally. @csomor I don't know if we/you want to change this?

**Comment by csomor** [comment]
Yes it's done on purpose. The comment above these lines cited refers to the HIG. IMHO if it's something critical, then the user should be offered an option (Yes or No, Ok or cancel) to prevent the consequences, then it's critical, then it get's the icon. 

That's following the guidelines in general and leads to the best platform impression in my view, avoiding the alert fatigue that comes with the overuse of these on other platforms, where these WARNING signs then tend to be clicked away by the user without even reading them properly, even if the would have been CRITICAL.

But if everybody else wants to match these signs 1:1 and not as platform customs indicate, then we can switch it.

**Comment by wouterbt** [comment]
I completely agree with the notion that the caution symbol should be used sparingly, as per the [HIG](https://developer.apple.com/design/human-interface-guidelines/alerts#macOS). However, I think that should be the _programmer's_ responsibility, and not be enforced by the GUI library. Furthermore, I do not think that having to make a choice (Yes / No) as opposed to only confirming the message (OK) should be the determining factor for whether or not to display the icon. I can think of instances where it is equally important that the user actually _read_ the message before continuing their work, even if there is no choice to make.

So I agree that we should change the code and perhaps add a remark to the documentation of the `wxICON_ERROR` style urging the programmer to adhere to the HIG.
